robot/server/server.go

```diff
@@ -322,20 +324,19 @@ func (s *Server) StreamStatus(req *pb.StreamStatusRequest, streamServer pb.Robot
 	ticker := time.NewTicker(every)
 	defer ticker.Stop()
 	for {
-		select {
-		case <-streamServer.Context().Done():
+		if !utils.SelectContextOrWaitChan(streamServer.Context(), ticker.C) {
 			return streamServer.Context().Err()
-		default:
-		}
-		select {
-		case <-streamServer.Context().Done():
-			return streamServer.Context().Err()
-		case <-ticker.C:
 		}
+
 		status, err := s.GetStatus(streamServer.Context(), &pb.GetStatusRequest{ResourceNames: req.ResourceNames})
-		if err != nil {
+		switch {
+		case err == nil:
+		case grpcstatus.Code(err) == codes.Unimplemented:
+			return nil
```

Comment:
If it's Unimplemented there's no sense in staying in this for loop, repeatedly making calls to `GetStatus` every tick.